cpp/src/quantiles/tdigest/tdigest_aggregation.cu

```diff
@@ -1161,6 +1174,10 @@ std::unique_ptr<column> merge_tdigests(tdigest_column_view const& tdv,
 
   auto const delta = max_centroids;
 
+  // We do not know whether there is any empty cluster in the input without actually reading the
+  // data, which could be expensive. So, we just assume that there could be empty clusters.
+  auto const may_have_empty_clusters = true;
```

Comment:
So we would never change it?

Reply:
Before this PR, this flag was fixed to false, which implies that empty clusters should never be found during the merge. This is not always true, which is the root cause of the bug. It is rather always true to assume that there might be some clusters until we inspect all clusters, which is expensive. Having this flag fixed to true might be always not optimal when there is indeed no empty cluster. We can make this better later for this case if that turns out to be a problem.